pkg/kvstore/store/store.go

```diff
@@ -0,0 +1,429 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/kvstore"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// listTimeoutDefault is the default timeout to wait while performing
+	// the initial list operation of objects from the kvstore
+	listTimeoutDefault = 30 * time.Second
+
+	// synchronizationIntervalDefault is the default interval to
+	// synchronize keys with the kvstore
+	synchronizationIntervalDefault = time.Minute
+
+	// watcherChanSize is the size of the channel to buffer kvstore events
+	watcherChanSize = 100
+)
+
+var (
+	controllers controller.Manager
+
+	log = logrus.WithField(logfields.LogSubsys, "shared-store")
```

Comment:
this should be `logging.DefaultLogger.WithField...` where logging is `pkg/logging`